### PR TITLE
[LUMI-133] Create Backend Package and drop mzai top level package 

### PR DESCRIPTION
**Pull request by chainlink**
## What's changing
This is a incremental PR of the the POC here: https://github.com/mozilla-ai/lumigator/pull/235

- Move backend one folder deeper to create a backend package inside the backend folder (enables us to continue to use absolute paths vs relative in the backend project once we migrate off pants)
- Drop `mzai` import path and make each top level module it's own independent package
- Temporarily move schemas inside backend to ease transition
- Update pants to run tests and build correctly.
- Update docker compose to reference new location.

Phases:
- Create backend package and prep  <-- We are here
- Add UV to backend project (upcoming PR)
- Add UV enabled backend dockerfile
- Update CI to build new backend image
- Move schemas back to it's own module + add UV + docker
- Make summarizer into a package and import Summarizerconfig (remove duplication hack) 
- Update Tests to run without pants
- Change CI to use non pants tests
- Run lint outside of pants
- Drop all pants files
- Tighten up folder structure (don't need it to be so deeply nested)

```
lumigator
    ├── backend (Dockerfile)
    │   └── __init__.py
    └   └── pyproject.toml
    ├── schemas 
    │   └── __init__.py
    └   └── pyproject.toml
    ├── config
    │   └── __init__.py
    └   └── pyproject.toml
    ├── evaluator
    │   └── __init__.py
    └   └── pyproject.toml
    ├── sdk (PIP-installable)
    │   └── __init__.py
        └── pyproject.toml
```

## How to test it

- [x] Test `make local-up`
- [x] Test pants Docker build
- [x] Tests passing

Test functionality (Local)
- [x] Upload File
- [x] Run Eval
- [x] Run Summarizer Deployment

## Related Jira Ticket
https://mzai.atlassian.net/browse/LUMI-133

## I already...

- [ ] added some tests for any new functionality;
- [ ] updated the documentation.

### CODE REVIEW

File: .devcontainer/Dockerfile.backend [hunk 1]
```diff
@@ -21,6 +21,5 @@ EOF
 
 
 WORKDIR /mzai
-ENV PYTHONPATH="/mzai"
```
Comment (chainlink) [inline .devcontainer/Dockerfile.backend:24]: These are extraneous (can be removed in current main as well)

File: lumigator/python/mzai/backend/backend/api/deployments/summarizer_config_loader.py [hunk 2]
```diff
@@ -4,10 +4,15 @@
...
 
-from mzai.backend.api.deployments.configloader import ConfigLoader
-from mzai.backend.settings import settings
-from mzai.summarizer.summarizer import SummarizerArgs
-
+from backend.api.deployments.configloader import ConfigLoader
+from backend.settings import settings
+
+# Note: this class is duplicated in lumigator/python/mzai/summarizer/summarizer.py
+class SummarizerArgs(BaseModel):
```
Comment (chainlink) [inline lumigator/python/mzai/backend/backend/api/deployments/summarizer_config_loader.py:11]: Duplicated this here for now (During Transition)

File: lumigator/BUILD.pants [hunk 3]
```diff
@@ -1,29 +0,0 @@
-workdir = "/lumigator/infra/mzai/docker"
```
Comment (chainlink) [inline lumigator/BUILD.pants:1]: General cleanup (this wasn't used)

File: .devcontainer/Dockerfile.backend [hunk 4]
```diff
@@ -21,6 +21,5 @@ EOF
...
-CMD zip lumigator/python/mzai/backend/api/deployments lumigator/python/mzai/backend/api/deployments.zip
```
Comment (veekaybee) [inline .devcontainer/Dockerfile.backend:25]: we still need to do this until we remove Ray Serve, I think

File: Makefile [hunk 5]
```diff
@@ -166,7 +166,7 @@ ci-fmt: ci-lint
 
 ci-tests:
 	@echo "running tests with parametrized platform: $(PARAMETRIZE)"
-	pants test lumigator/python/mzai/backend/tests:backend_tests@parametrize=$(PARAMETRIZE)
+	pants test lumigator/python/mzai/backend/backend/tests:backend_tests@parametrize=$(PARAMETRIZE)
```
Comment (veekaybee) [inline Makefile:169]: are there two levels of backend here?

**Comment by veekaybee** [comment]
Looks good as a first pass. Added a few comments and tested local app, top-level health routes all working and Docker also works.

**Review by veekaybee — APPROVED** [review]
Approved after some comments

File: .devcontainer/Dockerfile.backend (hunk 4 above)
Comment (chainlink) [reply .devcontainer/Dockerfile.backend:25]: Tested and working without this.

File: Makefile (hunk 5 above)
Comment (chainlink) [reply Makefile:169]: yes (confusingly) the inner backend is to make the 'backend.blah` python structure work. The outer `backend` contains the dockerfile, pyproject.toml, etc.

**Pull request merged**